index.bs

```diff
@@ -7,14 +7,110 @@ Group: browser-testing-tools
 URL: https://w3c.github.io/webdriver-bidi/
 No Editor: true
 Abstract: This document defines the BiDirectional WebDriver Protocol, a mechanism for remote control of user agents.
+Default Ref Status: current
+</pre>
+
+<pre class=anchors>
+spec: RFC6455; urlPrefix: https://tools.ietf.org/html/rfc6455
+    type: dfn
+        text: WebSocket ws-URI; url: section-3
+        text: A WebSocket Message Has Been Received; url: section-6.2
+        text: The WebSocket Closing Handshake is Started; url: section-7.1.3
+spec: WEBDRIVER; urlPrefix: https://w3c.github.io/webdriver/
+    type: dfn
+        text: additional capability deserialization algorithm; url: dfn-additional-capability-deserialization-algorithm
+        text: additional WebDriver capability; url: dfn-additional-webdriver-capability
+        text: capability name; url: dfn-capability-name
+        text: error code; url: dfn-error-code
+        text: error; url: dfn-errors
+        text: invalid argument; url: dfn-invalid-argument
+        text: matched capability serialization algorithm; url: dfn-matched-capability-serialization-algorithm
+        text: remote end; url: dfn-remote-ends
+        text: session; url: dfn-sessions
+        text: success; url: dfn-success
 </pre>
 
 Introduction {#intro}
 =====================
 
-[[WEBDRIVER|WebDriver]] defines a protocol for introspection and remote control
-of user agents. This specification extends WebDriver by introducing
-bidirectional communication. In place of the strict command/response
-format of WebDriver, this permits events to stream from the user
-agent to the controlling software, better matching the evented nature
-of the browser DOM.
+[[WEBDRIVER|WebDriver]] defines a protocol for introspection and
+remote control of user agents. This specification extends WebDriver by
+introducing bidirectional communication. In place of the strict
+command/response format of WebDriver, this permits events to stream
+from the user agent to the controlling software, better matching the
+evented nature of the browser DOM.
+
+Transport {#transport}
+======================
+
+Message transport is provided using a [[RFC6455|WebSocket]]
+connection.
+
+A [=remote end=] has a <dfn>connection</dfn>, which is a network connection, and
+a <dfn>websocket url</dfn>, which is a string. These are both initially null.
+
+When data is recieved on [=connection=] it must be processed according to the
+requirements of the [[!RFC6455|WebSockets]] specification. When [=a WebSocket
+message has been received=], an implementation must [=handle an incoming
+message=]. When [=the WebSocket closing handshake is started=] an implementation
+must [=end the session=].
+
+When required to <dfn>ensure a connection is established</dfn>, run
```

Comment:
I went with "start listening for a WebSocket connection", which I think is closer to the truth. Although I suppose it's kind of wrong in detail since there's actually a two-step process here 1. start listening on a port for new connections and 2. actually accept the connection. I guess I ought to do some more work to make this right :/